src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -353,34 +353,35 @@ ExecuteCascadeOperationForRelationIdList(List *relationIdList,
 	foreach_oid(relationId, relationIdList)
 	{
 		/*
-		 * Caller already passed the relations that we should operate on,
-		 * so we should not cascade here.
+		 * The reason behind skipping certain table types in below loop is
+		 * that we support some sort of foreign keys between postgres tables
+		 * and citus tables when enable_local_reference_table_foreign_keys is
+		 * false or when coordinator is not added to metadata.
+		 *
+		 * Also, as caller already passed the relations that we should operate
+		 * on, we don't cascade via foreign keys here.
 		 */
```

Comment:
It might be a bit nicer to move this comment into the undistribute table section or at least make it very clear that this refers to the undistribute table case.

Reply:
originally, this comment was in create citus local table part, but as the same also applies to undistribute table, I moved comment to here, in order to refer both cases